CCDB/src/CCDBDownloader.cxx

```diff
@@ -516,12 +516,18 @@ void CCDBDownloader::transferFinished(CURL* easy_handle, CURLcode curlCode)
         } else if (300 <= httpCode && httpCode < 400 && performData->locInd < locations.size()) {
           followRedirect(performData, easy_handle, locations, rescheduled, contentRetrieved);
         } else if (200 <= httpCode && httpCode < 300) {
-          contentRetrieved = true;
+          contentRetrieved = true; // Can be overruled by following timeout check
         }
       } else {
         LOG(error) << loggingMessage;
       }
 
+      // Check for timeout
+      if (curlCode == CURLE_OPERATION_TIMEDOUT) {
```

Comment:
Are there perhaps other error codes besides `CURLE_OPERATION_TIMEDOUT` for which we want to check?

Reply:
While I didn't find any additional error codes that would prompt wiping the destination vector, the `transferFinished` function certainly lacks a general error check which I've added in the following commit.